src/main/java/com/artipie/YamlSettings.java

```diff
@@ -86,7 +86,7 @@ public Storage storage() throws IOException {
     public CompletionStage<Authentication> auth() {
         return this.credentials().thenCompose(
             opt -> opt.<CompletionStage<Authentication>>map(
-                creds -> creds.yaml().thenApply(AuthFromYaml::new)
+                creds -> ((Credentials.FromStorageYaml) creds).yaml().thenApply(AuthFromYaml::new)
```

Comment:
@olenagerasimova casting `creds` to `Credentials.FromStorageYaml` might cause problems in the future due to types mismatches in runtime. I have two solutions in mind:
- Signature of `CompletionStage<Optional<Credentials>> credentials()` method may be changed to `CompletionStage<Optional<Credentials.FromStorageYaml>> credentials()` in this class. `Settings` interface remains untouched.
- `CompletionStage<Authentication> auth()` method may be added to `Credentials` interface and creating `AuthFromYaml` is moved to `auth()` implementation in `Credentials.FromStorageYaml`.
What do you think?

Reply:
@olegmoz hm, I like the second option better, it will be more flexible and handy when we will have other `Authentication` and `Credentials` implementations. And technically we have credentials from env now, we can create corresponding `Credentials` implementation letter. 
Do you mind if I todo your suggestion?

Reply:
@olenagerasimova yes, adding todo will be fine. Please mention removing this dynamic cast in the puzzle.